pyfr/backends/openmp/kernels/errest.mako

```diff
@@ -7,30 +7,58 @@ errest(int nrow, int ncolb, int ldim, fpdtype_t *__restrict__ error,
        fpdtype_t *__restrict__ x, fpdtype_t *__restrict__ y,
        fpdtype_t *__restrict__ z, fpdtype_t atol, fpdtype_t rtol)
 {
-    #define X_IDX_AOSOA(v, nv) ((ci/SOA_SZ*(nv) + (v))*SOA_SZ + cj)
+    int nblocks = ncolb/SZ;
+    #define X_IDX_AOSOA(v, nv) ((_xi/SOA_SZ*(nv) + (v))*SOA_SZ + _xj)
 
     // Initalise the reduction variables
     fpdtype_t ${','.join('err{0} = 0.0'.format(i) for i in range(ncola))};
 
+% if norm == 'uniform':
+    #pragma omp parallel for reduction(max : ${','.join('err{0}'.format(i) for i in range(ncola))})
+% else:
+    #pragma omp parallel for reduction(+ : ${','.join('err{0}'.format(i) for i in range(ncola))})
+% endif
+    for ( int ib = 0; ib < nblocks; ib++ )
+    {
+        int idx;
+
+        for ( int _xi = 0; _xi < SZ; _xi += SOA_SZ )
+        {
+            for ( int _y = 0; _y < nrow; _y++ )
```

Comment:
I think we should interchange the xi and y loop orders

Reply:
We need xi loop before y loop in most of the pointwise kernels (ones generated as usual) but errest and axnpby is that right?

Reply:
We have freedom to interchange them (at least from a correctness standpoint). I believe performance will be better if we do interchange.